src/transform/src/map_lifting.rs

```diff
@@ -191,13 +273,49 @@ impl LiteralLifting {
 
                 if scalars.is_empty() {
                     *relation = input.take_dangerous();
+                } else {
+                    // Permute columns to put literals at end, if any, hope project lifted.
+                    let literal_count = scalars.iter().filter(|e| e.is_literal()).count();
+                    if literal_count != 0 {
+                        let input_arity = input.arity();
+                        let first_literal_id = input_arity + scalars.len() - literal_count;
+                        let mut new_scalars = Vec::new();
+                        let mut projected_literals = Vec::new();
+                        let mut projection = (0..input_arity).collect::<Vec<usize>>();
+                        let mut column_map = projection.clone();
```

Comment:
It looks like `column_map` is always identical to `projection` and thus unnecessary.

Reply:
You're absolutely right.